### PR TITLE
fix: ButtonGroup `Each child in a list should have a unique "key" prop`

### PR DESCRIPTION
**Pull request by 0x30**
`ButtonGroup` split line `<span>`  add key.

### What kind of change does this PR introduce? (check at least one)

 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update
 - [ ] Refactor
 - [ ] Test Case
 - [ ] TypeScript definition update
 - [ ] Document improve
 - [ ] CI/CD improve
 - [ ] Branch sync
 - [ ] Other, please describe:

### PR description

1. `ButtonGroup` 组件在增加分割的竖线时，由于没有设置 `key` 触发了警报
2. 关于 `ButtonGroup` 的会覆盖`children` 的 key 值，不认为这样是好的方式，应尊重用户的第一选择，如果没有可以使用 index 作为 `key`

### Changelog
🇨🇳 Chinese
- Fix: 修复 `Warning: Each child in a list should have a unique "key" prop.`

image

### Checklist
- [x] Test
- [ ] Document or no need
- [ ] Changelog or no need

### Other
- [ ] Skip Changelog

### CODE REVIEW

File: packages/semi-ui/button/buttonGroup.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ export default class ButtonGroup extends BaseComponent<ButtonGroupProps> {
         if (children) {
             inner = ((Array.isArray(children) ? children : [children])).map((itm: React.ReactNode, index) => (
                 isValidElement(itm)
-                    ? cloneElement(itm, { disabled, size, type, ...itm.props, ...rest, key: index })
+                    ? cloneElement(itm, { disabled, size, type, ...itm.props, ...rest, key: itm.props.key ?? index })
```
Comment (pointhalo) [inline packages/semi-ui/button/buttonGroup.tsx:83]: item 是 reactNode，key是特殊属性。是无法通过  itm.props.key 这种方式取到的。这里的写法不太对。

Comment (pointhalo) [reply packages/semi-ui/button/buttonGroup.tsx:83]: 需要改一下。

Comment (0x30) [reply packages/semi-ui/button/buttonGroup.tsx]: 好的 🫡

**Comment by pointhalo** [comment]
release v2.20.3

**Pull request merged**